zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -4796,6 +4834,60 @@ components:
         incidentKey:
           description: Incident key associated with this flow node instance.
           type: string
+    AdHocSubprocessActivitySearchQuery:
+      description: Ad-hoc subprocess activities search request.
+      type: object
+      required:
+        - processDefinitionKey
+        - adHocSubprocessId
+      properties:
+        processDefinitionKey:
+          description: The process definition key.
+          type: string
+        adHocSubprocessId:
+          description: The ad-hoc subprocess process element ID.
+          type: string
+    AdHocSubprocessActivitySearchQueryResult:
```

Comment:
As we're not doing a real search I opted for not including paging/sorting on this API. Open to discuss if we want this in terms of consistency, but I guess this would make the implementation more complex as we're not relying on the datastore here and would need to implement this directly.